GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -238,25 +240,31 @@ GDTCORNetworkMobileSubtype GDTCORNetworkMobileSubTypeMessage() {
   return resultData;
 }
 
+id<NSSecureCoding> _Nullable GDTCORDecodeArchiveAtPath(Class archiveClass,
+                                                       NSString *_Nonnull archivePath,
+                                                       NSError **_Nonnull error) {
+  NSData *data = [NSData dataWithContentsOfFile:archivePath options:0 error:error];
+  if (data == nil) {
+    // Reading the file failed and `error` will be populated.
+    return nil;
+  }
+
+  return GDTCORDecodeArchive(archiveClass, data, error);
+}
+
 id<NSSecureCoding> _Nullable GDTCORDecodeArchive(Class archiveClass,
-                                                 NSString *_Nullable archivePath,
-                                                 NSData *_Nullable archiveData,
-                                                 NSError *_Nullable *error) {
+                                                 NSData *_Nonnull archiveData,
+                                                 NSError **_Nonnull error) {
   id<NSSecureCoding> unarchivedObject = nil;
   if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, watchOS 4, *)) {
-    NSData *data = archiveData ? archiveData : [NSData dataWithContentsOfFile:archivePath];
```

Comment:
A note here on why this API changed...
This was a robust but tricky API because it allowed for the client to either pass in a file path to decode or some data to decode. While the API fulfilled two use cases, I discovered that it was silently failing when passing in a file path to a resource that does not exist. The `[NSData dataWithContentsOfFile:archivePath]` API used to decode the file path does not provide a hook to pass in an error and therefore pass back the "file DNE" error. This broke the result type semantics of the API as, in the case of the file resource not existing, a nil object was returned and the error ptr was not populated.

I thought it better to just split the decoding API into two where one is explicitly for decoding a file resource and the other is for decoding data in memory. I used `_Nonnull` to highlight invalid states and encourage the client to use the error param.